[32mINFO    [0m integration.test_upgrade_data_integrator:test_upgrade_data_integrator.py:95 Refresh the charm
[32mINFO    [0m integration.test_upgrade_data_integrator:test_upgrade_data_integrator.py:98 Wait for upgrade to start
[32mINFO    [0m integration.test_upgrade_data_integrator:test_upgrade_data_integrator.py:105 Wait for upgrade to complete on first upgrading unit
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  pgbouncer-k8s/0 [executing] waiting: other units upgrading first...
  pgbouncer-k8s/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  pgbouncer-k8s/0 [idle] waiting: other units upgrading first...
  pgbouncer-k8s/1 [idle] active: 
[32mINFO    [0m integration.test_upgrade_data_integrator:test_upgrade_data_integrator.py:113 Resume upgrade
[32mINFO    [0m integration.test_upgrade_data_integrator:test_upgrade_data_integrator.py:118 Wait for upgrade to complete
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  pgbouncer-k8s/0 [executing] maintenance: stopping charm software
  pgbouncer-k8s/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  pgbouncer-k8s/0 [idle] active: 
  pgbouncer-k8s/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  pgbouncer-k8s/0 [idle] active: 
  pgbouncer-k8s/1 [idle] active: 
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  pgbouncer-k8s/0 [idle] active: 
  pgbouncer-k8s/1 [idle] active:
[32mINFO    [0m pytest_operator.plugin:plugin.py:903 Model status:

Model  Controller          Cloud/Region        Version  SLA          Timestamp
test   microk8s-localhost  microk8s/localhost  3.4.5    unsupported  01:43:10Z

App              Version  Status  Scale  Charm            Channel      Rev  Address         Exposed  Message
data-integrator           active      2  data-integrator  latest/edge   41  10.152.183.53   no       
pgbouncer-k8s    1.21.0   active      2  pgbouncer-k8s                   0  10.152.183.229  no       
postgresql-k8s   14.12    active      3  postgresql-k8s   14/edge      349  10.152.183.109  no       

Unit                Workload  Agent  Address     Ports  Message
data-integrator/0*  active    idle   10.1.58.72         
data-integrator/1   active    idle   10.1.58.73         
pgbouncer-k8s/0*    active    idle   10.1.58.83         
pgbouncer-k8s/1     active    idle   10.1.58.82         
postgresql-k8s/0    active    idle   10.1.58.79         
postgresql-k8s/1*   active    idle   10.1.58.81         Primary
postgresql-k8s/2    active    idle   10.1.58.80         

[32mINFO    [0m pytest_operator.plugin:plugin.py:909 Juju error logs:

unit-postgresql-k8s-2: 01:35:06 ERROR unit.postgresql-k8s/2.juju-log Uncaught exception while in charm code:
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-k8s-2/charm/venv/ops/model.py", line 3187, in _run
    result = subprocess.run(args, **kwargs)  # type: ignore
  File "/usr/lib/python3.10/subprocess.py", line 526, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '('/var/lib/juju/tools/unit-postgresql-k8s-2/secret-get', '--label', 'database-peers.postgresql-k8s.app', '--format=json')' returned non-zero exit status 1.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-k8s-2/charm/./src/charm.py", line 2071, in <module>
    main(PostgresqlOperatorCharm, use_juju_for_storage=True)
  File "/var/lib/juju/agents/unit-postgresql-k8s-2/charm/venv/ops/main.py", line 551, in main
    manager.run()
  File "/var/lib/juju/agents/unit-postgresql-k8s-2/charm/venv/ops/main.py", line 530, in run
    self._emit()
  File "/var/lib/juju/agents/unit-postgresql-k8s-2/charm/venv/ops/main.py", line 519, in _emit
    _emit_charm_event(self.charm, self.dispatcher.event_name)
  File "/var/lib/juju/agents/unit-postgresql-k8s-2/charm/venv/ops/main.py", line 147, in _emit_charm_event
    event_to_emit.emit(*args, **kwargs)
  File "/var/lib/juju/agents/unit-postgresql-k8s-2/charm/venv/ops/framework.py", line 348, in emit
    framework._emit(event)
  File "/var/lib/juju/agents/unit-postgresql-k8s-2/charm/venv/ops/framework.py", line 860, in _emit
    self._reemit(event_path)
  File "/var/lib/juju/agents/unit-postgresql-k8s-2/charm/venv/ops/framework.py", line 950, in _reemit
    custom_handler(event)
  File "/var/lib/juju/agents/unit-postgresql-k8s-2/charm/lib/charms/tempo_k8s/v1/charm_tracing.py", line 647, in wrapped_function
    return callable(*args, **kwargs)  # type: ignore
  File "/var/lib/juju/agents/unit-postgresql-k8s-2/charm/./src/charm.py", line 868, in _on_postgresql_pebble_ready
    self.unit.set_workload_version(self._patroni.rock_postgresql_version)
  File "/var/lib/juju/agents/unit-postgresql-k8s-2/charm/./src/charm.py", line 1449, in _patroni
    self.get_secret(APP_SCOPE, REPLICATION_PASSWORD_KEY),
  File "/var/lib/juju/agents/unit-postgresql-k8s-2/charm/lib/charms/tempo_k8s/v1/charm_tracing.py", line 647, in wrapped_function
    return callable(*args, **kwargs)  # type: ignore
  File "/var/lib/juju/agents/unit-postgresql-k8s-2/charm/./src/charm.py", line 312, in get_secret
    if result := self.peer_relation_data(scope).fetch_my_relation_field(peers.id, key):
  File "/var/lib/juju/agents/unit-postgresql-k8s-2/charm/lib/charms/data_platform_libs/v0/data_interfaces.py", line 1260, in fetch_my_relation_field
    if relation_data := self.fetch_my_relation_data([relation_id], [field], relation_name):
  File "/var/lib/juju/agents/unit-postgresql-k8s-2/charm/lib/charms/data_platform_libs/v0/data_interfaces.py", line 1249, in fetch_my_relation_data
    data[relation.id] = self._fetch_my_specific_relation_data(relation, fields)
  File "/var/lib/juju/agents/unit-postgresql-k8s-2/charm/lib/charms/data_platform_libs/v0/data_interfaces.py", line 534, in wrapper
    return f(self, *args, **kwargs)
  File "/var/lib/juju/agents/unit-postgresql-k8s-2/charm/lib/charms/data_platform_libs/v0/data_interfaces.py", line 2083, in _fetch_my_specific_relation_data
    self.component, self.secret_fields, relation, fields
  File "/var/lib/juju/agents/unit-postgresql-k8s-2/charm/lib/charms/data_platform_libs/v0/data_interfaces.py", line 1817, in secret_fields
    self.static_secret_fields if self.static_secret_fields else self.current_secret_fields
  File "/var/lib/juju/agents/unit-postgresql-k8s-2/charm/lib/charms/data_platform_libs/v0/data_interfaces.py", line 1836, in current_secret_fields
    if content := self._get_group_secret_contents(relation, group):
  File "/var/lib/juju/agents/unit-postgresql-k8s-2/charm/lib/charms/data_platform_libs/v0/data_interfaces.py", line 2070, in _get_group_secret_contents
    result = super()._get_group_secret_contents(relation, group, secret_fields)
  File "/var/lib/juju/agents/unit-postgresql-k8s-2/charm/lib/charms/data_platform_libs/v0/data_interfaces.py", line 1000, in _get_group_secret_contents
    if (secret := self._get_relation_secret(relation.id, group)) and (
  File "/var/lib/juju/agents/unit-postgresql-k8s-2/charm/lib/charms/data_platform_libs/v0/data_interfaces.py", line 506, in wrapper
    return f(self, *args, **kwargs)
  File "/var/lib/juju/agents/unit-postgresql-k8s-2/charm/lib/charms/data_platform_libs/v0/data_interfaces.py", line 2059, in _get_relation_secret
    return self.secrets.get(label, secret_uri, legacy_labels=self._previous_labels())
  File "/var/lib/juju/agents/unit-postgresql-k8s-2/charm/lib/charms/data_platform_libs/v0/data_interfaces.py", line 706, in get
    if secret.meta:
  File "/var/lib/juju/agents/unit-postgresql-k8s-2/charm/lib/charms/data_platform_libs/v0/data_interfaces.py", line 605, in meta
    self._secret_meta = self._model.get_secret(label=label)
  File "/var/lib/juju/agents/unit-postgresql-k8s-2/charm/venv/ops/model.py", line 291, in get_secret
    content = self._backend.secret_get(id=id, label=label)
  File "/var/lib/juju/agents/unit-postgresql-k8s-2/charm/venv/ops/model.py", line 3552, in secret_get
    result = self._run('secret-get', *args, return_output=True, use_json=True)
  File "/var/lib/juju/agents/unit-postgresql-k8s-2/charm/venv/ops/model.py", line 3189, in _run
    raise ModelError(e.stderr) from e
ops.model.ModelError: ERROR permission denied

unit-postgresql-k8s-2: 01:35:06 ERROR juju.worker.uniter.operation hook "postgresql-pebble-ready" (via hook dispatching script: dispatch) failed: exit status 1
unit-postgresql-k8s-2: 01:35:06 ERROR juju.worker.uniter pebble poll failed for container "postgresql": failed to send pebble-ready event: hook failed

[32mINFO    [0m pytest_operator.plugin:plugin.py:991 Forgetting model main...